src/app/components/filters/filters.test.tsx

```diff
@@ -0,0 +1,96 @@
+import { AdditionalFilters, Stops } from "../../models/flight-parameters";
+import Filters, { FiltersProps } from "./filters";
+import { PriceRangeFilterProps } from "./price-range-filter/price-range-filter";
+import { StopsFilterProps } from "./stops-filter/stops-filter";
+import { fireEvent, render, screen } from "@testing-library/react";
+
+enum MockedComponentTestIds {
+  StopsFilter = 'stops-filter',
+  PriceRangeFilterProps = 'price-range-filter'
+}
+
+jest.mock('./filters.css', () => {});
+
+jest.mock('../../../components/ui/card', () => ({
+  Card: ({ children }: any) => <div>{...children}</div>
+}));
+
+jest.mock('./stops-filter/stops-filter', () => ({
+  StopsFilter: ({ activeFilter, setActiveFilter }: StopsFilterProps) => {
+    return <button data-testid={MockedComponentTestIds.StopsFilter}
+      onClick={() => setActiveFilter(Stops.NonStop)}>{activeFilter}</button>
+  }
+}));
+
+jest.mock('./price-range-filter/price-range-filter', () => ({
+  PriceRangeFilter: (props: PriceRangeFilterProps) => {
+    return <button data-testid={MockedComponentTestIds.PriceRangeFilterProps}
+      onClick={() => props.setSelectedRange({ min: props.minOnScale + 5, max: props.maxOnScale })}></button>
+  }
+}))
+
+describe(Filters.name, () => {
+  let props: FiltersProps;
+  beforeEach(() => {
+    props = {
+      min: 200,
+      max: 300,
+      setFilters: jest.fn()
+    };
+
+    jest.clearAllMocks();
+  });
+
+  it('should not render PriceRange filter component if min is not defined', () => {
+    const propsWithMinUndefined = { ...props, min: undefined };
+
+    render(<Filters {...propsWithMinUndefined}></Filters>);
+
+    const priceRangeFilterComponent = screen.queryByTestId(MockedComponentTestIds.PriceRangeFilterProps);
+    expect(priceRangeFilterComponent).toBeNull();
```

Comment:
not to be in document?

Reply:
the extended assertion from jest-dom were not showing up, there needs to be some configuration setup done. Did try to have them, but then moved away to concentrate on tests